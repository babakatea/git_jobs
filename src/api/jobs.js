import axios from "axios";

const url = 'http://localhost:3000/api/get_jobs/';

async function fetchJobs() {
    const {data} = await axios.get(url);

    return data;
}

export {fetchJobs};
