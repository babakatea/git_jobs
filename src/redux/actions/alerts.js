import {alertActionsTypes} from '../constants/index';

function success(message) {
  return {type: alertActionsTypes.ALERT_SUCCESS, message};
}

function error(message) {
  return {type: alertActionsTypes.ALERT_ERROR, message};
}

function hide() {
  return {type: alertActionsTypes.ALERT_HIDE};
}

export default {success, error, hide};
