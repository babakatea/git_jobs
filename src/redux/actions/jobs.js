import services from '../services';
import {jobsActionTypes} from "../constants";

const success_list_jobs = (list) => ({
  type: jobsActionTypes.JOBS_LIST_LOADED,
  list
});

const fail_list_jobs = (error) => ({
  type: jobsActionTypes.JOBS_LIST_LOAD_FAILED, error
});

const success_details_job = (details) => ({
  type: jobsActionTypes.JOBS_DETAILS_LOADED,
  details
});

const fail_details_job = (error) => ({
  type: jobsActionTypes.JOBS_DETAILS_LOAD_FAILED,
  error
});

const request_list_jobs = () => ({
  type: jobsActionTypes.JOBS_LIST_REQUEST,
});

const request_job_details = () => ({
  type: jobsActionTypes.JOBS_DETAILS_LOAD_REQUESTED,
});

export const loadJobsList = (params) => {
  return dispatch => {
    dispatch(request_list_jobs());
    return services.getJobs(params).then(data => {
      dispatch(success_list_jobs(data));
    }, error => {
      dispatch(fail_list_jobs(error));
    });
  };
};

export const loadJobsDetails = (params) => {
  return dispatch => {
    dispatch(request_job_details());
    return services.getDetails(params).then(data => {
      dispatch(success_details_job(data));
    }, error => {
      dispatch(fail_details_job(error));
    });
  };
};
