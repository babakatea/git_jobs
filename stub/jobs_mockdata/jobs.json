[
  {
    "id": 1,
    "name": "Frontend developer",
    "location": "Los Angeles, US",
    "salary": 100000
  },
  {
    "id": 2,
    "name": "Accountant",
    "location": "Moscow, US",
    "salary": 50000
  },
  {
    "id": 3,
    "name": "Frontend developer",
    "location": "Los Angeles, US",
    "salary": 100000
  },
  {
    "id": 4,
    "name": "DevOps developer",
    "location": "Innopolis",
    "salary": 12000
  },
  {
    "id": 5,
    "name": "Backend developer",
    "location": "Kazan, Russia",
    "salary": 20000
  },
  {
    "id": 6,
    "name": "Frontend developer",
    "location": "Los Angeles, US",
    "salary": 100000
  },
  {
    "id": 7,
    "name": "DevOps developer",
    "location": "Innopolis",
    "salary": 12000
  },
  {
    "id": 8,
    "name": "Backend developer",
    "location": "Kazan, Russia",
    "salary": 20000
  }
]