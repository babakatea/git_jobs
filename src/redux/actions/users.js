import {userActionsTypes} from '../constants';
import services from '../services';
import history from '../../history';
import alerts from './alerts';


function login(email, password) {
  return dispatch => {
    dispatch(request());

    const token = services.login(email, password);

    if (token) {
      localStorage.setItem('token', token);
      history.push('/search');
      dispatch(success({token: token, loggedIn: true}));
    } else {
      dispatch(failure("Failed to login"));
      dispatch(alerts.error("Failed to login"));
    }
  };

  function request() {
    return {type: userActionsTypes.USERS_LOGIN_REQUEST}
  }

  function success(payload) {
    return {type: userActionsTypes.USERS_LOGIN_SUCCESS, payload}
  }

  function failure(error) {
    return {type: userActionsTypes.USERS_LOGIN_FAILURE, error}
  }
}

function getProfile() {
  return dispatch => {
    dispatch(request());
    services.getProfile()
      .then(
        user => dispatch(success(user)),
        error => {
          dispatch(failure(error));
          dispatch(alerts.error(error))
        }
      );
  };

  function request() {
    return {type: userActionsTypes.USERS_PROFILE_REQUEST}
  }

  function success(data) {
    return {type: userActionsTypes.USERS_LOGIN_SUCCESS, data}
  }

  function failure(error) {
    return {type: userActionsTypes.USERS_LOGIN_FAILURE, error}
  }
}

function logout() {
  services.logout();
  history.push('/');
  return {type: userActionsTypes.USERS_LOGOUT};
}

export default {login, logout, getProfile};
