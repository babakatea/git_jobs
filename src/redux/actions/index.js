import alerts from './alerts';
import users from './users';
import {loadJobsDetails, loadJobsList} from './jobs';

export default {alerts, users, jobs: {loadJobsDetails, loadJobsList}};
