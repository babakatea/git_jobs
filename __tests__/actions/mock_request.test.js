import moxios from 'moxios';
import {loadJobsList} from "../../src/redux/actions/jobs";
import React from "react";
const dispatch = jest.fn();

describe('action load jobs list', () => {

  beforeEach(() => {
    moxios.install()

  });

  it('load jobs with moxios', (done) => {

    const API_URL = `https://cors-anywhere.herokuapp.com/https://jobs.github.com/positions.json`;

    moxios.stubRequest(API_URL, {
      status: 200,
      responseText: {
        results: [{test: 'jobs'}]
      }
    });

    const params = {
      params: {}
    };

    const dispatcher = loadJobsList(params);
    expect(typeof dispatcher).toBe('function');

    dispatcher(dispatch);

    moxios.wait(function () {
      const request = moxios.requests.mostRecent();
      expect(dispatch).toBeCalledTimes(1);
      expect(dispatch.mock.calls[0][0]).toMatchSnapshot();
      done()
    })
  });

  afterEach(() => {
    moxios.uninstall()
  });

});
