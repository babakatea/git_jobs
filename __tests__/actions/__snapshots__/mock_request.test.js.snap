// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action load jobs list load jobs with moxios 1`] = `
Object {
  "type": "JOBS_LIST_REQUEST",
}
`;
